src/openms/source/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.cpp

```diff
@@ -261,45 +302,72 @@ namespace OpenMS
+          for (const auto& hit : ident.getHits())
           {
-            OPENMS_LOG_ERROR << e.what();
+            OpenMS::Feature f;
+            OpenMS::Feature s = feature;
+            f.setUniqueId();
+            s.setUniqueId();
+            if (hit.getMetaValue("identifier").toStringList().at(0) != "null")
+            {
+              f.setMetaValue("PeptideRef", hit.getMetaValue("identifier").toStringList().at(0));
+              s.setMetaValue("PeptideRef", hit.getMetaValue("identifier").toStringList().at(0));
+              std::string native_id = hit.getMetaValue("chemical_formula").toString() + ";" + hit.getMetaValue("modifications").toString();
+              s.setMetaValue("native_id", native_id);
+              s.setMetaValue("identifier", hit.getMetaValue("identifier"));
+              s.setMetaValue("description", hit.getMetaValue("description"));
+              s.setMetaValue("modifications", hit.getMetaValue("modifications"));
+              std::string adducts;
+              try
+              {
+                std::string str = hit.getMetaValue("modifications").toString();
+                std::string delimiter = ";";
+                adducts = str.substr(1, str.find(delimiter) - 1);
+              }
+              catch (const std::exception& e)
+              {
+                OPENMS_LOG_ERROR << e.what();
+              }
+              s.setMetaValue("adducts", adducts);
+              OpenMS::EmpiricalFormula chemform(hit.getMetaValue("chemical_formula").toString());
+              double adduct_mass = s.getMZ() * std::abs(hit.getCharge()) + static_cast<double>(hit.getMetaValue("mz_error_Da")) - chemform.getMonoWeight();
+              s.setMetaValue("dc_charge_adduct_mass", adduct_mass);
+              s.setMetaValue("chemical_formula", hit.getMetaValue("chemical_formula"));
+              s.setMetaValue("mz_error_ppm", hit.getMetaValue("mz_error_ppm"));
+              s.setMetaValue("mz_error_Da", hit.getMetaValue("mz_error_Da"));
+              s.setCharge(hit.getCharge());
+              f.setSubordinates({s});
+              feat_map_output.push_back(f);
+            }
+            else if (add_unidentified_features)
+            {
+              //// "PeptideRef" metavalue should have been set during peak picking, but if not...
+              std::ostringstream mass_of_the_peak;
+              mass_of_the_peak << s.getMZ();
+
+              // Fill in accurateMassSearch metavalues
+              DataValue identifiers(std::vector<std::string>({mass_of_the_peak.str()}));
+              s.setMetaValue("identifier", identifiers);
+              s.setMetaValue("description", "");
+              s.setMetaValue("modifications", "");
+              s.setMetaValue("adducts", "");
+              s.setMetaValue("dc_charge_adduct_mass", 0);
+              s.setMetaValue("chemical_formula", "");
+              s.setMetaValue("mz_error_ppm", 0);
+              s.setMetaValue("mz_error_Da", 0);
```

Comment:
do you really need to fill metavalues which do not exist?
Especially `mz_error_Da = 0` could easily be misinterpreted downstream.
Better to omit them entirely?

Reply:
The reasoning was to maintain these MetaValues for compatibility with the identified features but filled with "Null" values that downstream algorithms could then use to infer that the step was ran but no information is available for the MetaValues.

I definitely see your point about the case of `mz_error_Da = 0` and agree this is not optimal.  What would you recommend for that situation?  Something like `-1`? or something different altogether?

Reply:
Another option would be to add this algorithm as a DataProcessing step or a metavalue at the FeatureMap or Experiment level. This, plus a missing metavalue in the feature would then be enough.
But if you want you can think about it and we merge this PR first.